diff --git a/src/cli/Logger.cpp b/src/cli/Logger.cpp
index 23d0ee9..24e01c5 100644
--- a/src/cli/Logger.cpp
+++ b/src/cli/Logger.cpp
@@ -2,6 +2,7 @@
 
 #include <chrono>
 #include <fcntl.h>
+#include <version>
 #include <print>
 
 using namespace Hyprutils;
@@ -58,7 +59,9 @@ std::expected<void, std::string> CLogger::setOutputFile(const std::string_view&
     if (!m_impl->m_logOfs.good())
         return std::unexpected("Failed to open a write stream");
 
+#if defined(__cpp_lib_fstream_native_handle)
     fcntl(m_impl->m_logOfs.native_handle(), F_SETFD, FD_CLOEXEC);
+#endif
 
     m_impl->m_fileEnabled = true;
     m_impl->updateParentShouldLog();
