diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6c538d4..4498695 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,6 +128,7 @@ endif()
 configure_file(data/hwdata.hpp.in hwdata.hpp @ONLY)
 
 # tests
+if(BUILD_TESTING)
 add_custom_target(tests)
 
 add_executable(simpleWindow "tests/SimpleWindow.cpp")
@@ -145,6 +146,7 @@ add_test(
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/tests
   COMMAND attachments "attachments")
 add_dependencies(tests attachments)
+endif()
 
 # Installation
 install(TARGETS aquamarine)
