diff --git a/src/Compositor.cpp b/src/Compositor.cpp
index a4acbec..49d429b 100644
--- a/src/Compositor.cpp
+++ b/src/Compositor.cpp
@@ -311,7 +311,9 @@ void CCompositor::initServer(std::string socketName, int socketFd) {
         Debug::trace = true;
 
     // set the buffer size to 1MB to avoid disconnects due to an app hanging for a short while
+#if WAYLAND_VERSION_MAJOR > 1 || (WAYLAND_VERSION_MAJOR == 1 && WAYLAND_VERSION_MINOR >= 23)
     wl_display_set_default_max_buffer_size(m_sWLDisplay, 1_MB);
+#endif
 
     Aquamarine::SBackendOptions options{};
     options.logFunction = aqLog;
